jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/ResourceHttpContent.java

```diff
@@ -119,13 +119,13 @@ public HttpField getETag()
     @Override
     public String getETagValue()
     {
-        return _resource.getWeakETag();
+        return EtagUtil.calcWeakEtag(_resource.getPath());
```

Comment:
This would only work with path-backed resources. Shouldn't we be capable of calculating the etag of a potential `MemoryResource`?

Reply:
Fixed in this PR.